### PR TITLE
Enable DHCP and DHCPS implemented in LWIP

### PR DESCRIPTION
**Pull request by SeonghoByeon**
1. os/net/lwip: Register callback function for dhcps start
    - Add a global variable pointing out staion join callback in dhcps
    - Register callback using input from dhcps_start function
    - Modify corresponding examples

2. framework/wifi_manager: Enable LWIP's DHCP/DHCPS instead of external feature
    - Add dhcpc and dhcps operations for LWIP's DHCP
    - Add configuration settings for enabling/disabling LWIP's DHCP

3. os/net/lwip: Arrange DHCP configuration and modify DHCPS
    - Add LWIP DHCP/DHCPS configuration to be enabled or disabled
    - Modify DHCPS to differentiate in_addr_t structure depending on ipv6
    - Arrange Kconfig

### CODE REVIEW

**Comment by SeonghoByeon** [comment]
Hi, @pillip8282, @sangwon03, @JeonginKim.
I have successfully finished testing of DHCP/DHCPS of LWIP for both artik053 and cy4390x.
(unit testcase, stress test, and traffic test using iperf for all cases.)
By enabling this, Wi-Fi Manager is able to use DHCP/DHCPS of LWIP instead of netutils external one. 
Please let me know your opinion.
Thanks.

**Comment by seinfra** [comment]
Target : [99cb3dca24e728ac2aecaa58b5605d51f1273948] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [99cb3dca24e728ac2aecaa58b5605d51f1273948] - Code Rule Check Result:
framework/src/wifi_manager/wifi_manager.c:671: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxW)
framework/src/wifi_manager/wifi_manager.c:678: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/wifi_manager/wifi_manager.c:705: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxW)
framework/src/wifi_manager/wifi_manager.c:731: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxW)
framework/src/wifi_manager/wifi_manager.c:770: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxW)
framework/src/wifi_manager/wifi_manager.c:814: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxW)

**Comment by seinfra** [comment]
Target : [3480f5a4b54900b5764dcaa2f184b8db8d808239] - Code Rule Check Result:
framework/src/wifi_manager/wifi_manager.c:678: ERROR: [SPC_M_OPR] spaces required around that '<' (ctx:OxW)
framework/src/wifi_manager/wifi_manager.c:680: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:ExO)
framework/src/wifi_manager/wifi_manager.c:680: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:OxO)
framework/src/wifi_manager/wifi_manager.c:680: ERROR: [SPC_M_OPR] spaces required around that '==' (ctx:OxO)
framework/src/wifi_manager/wifi_manager.c:680: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:OxE)
framework/src/wifi_manager/wifi_manager.c:681: ERROR: [SPC_M_KWD] space required before the open parenthesis '('
framework/src/wifi_manager/wifi_manager.c:682: ERROR: [SPC_M_OPR] spaces required around that '>' (ctx:OxW)
framework/src/wifi_manager/wifi_manager.c:682: ERROR: [SPC_M_OPR] spaces required around that ':' (ctx:VxW)

File: framework/src/wifi_manager/wifi_manager.c [hunk 1]
```diff
@@ -654,29 +666,82 @@ wifi_manager_result_e _start_dhcpd(void)
...
+	struct netif *cur_netif;
+	cur_netif = netif_find(WIFIMGR_SOFTAP_IFNAME);
+	if (cur_netif == NULL) {
+		ndbg("[WM] DHCP Server - no network interface for dhcpd start\n");
+		WIFIADD_ERR_RECORD(ERR_WIFIMGR_CONNECT_DHCPD_FAIL);
+		return wret;
+	}
+	netif_set_up(cur_netif);
+<<<<<<< HEAD
+	if (dhcps_start(cur_netif) == ERR_OK) {
```
Comment (yashwanth686007) [inline framework/src/wifi_manager/wifi_manager.c]: I think this is merge conflict ?

**Comment by seinfra** [comment]
Target : [3480f5a4b54900b5764dcaa2f184b8db8d808239] - Code Rule Check (C++) OK.

File: framework/src/wifi_manager/wifi_manager.c (hunk 1 above)
Comment (SeonghoByeon) [reply framework/src/wifi_manager/wifi_manager.c]: Two commits are merged by mistake. This PR will be recreated. Thanks.

**Pull request closed without merging**